frontend/src/AuthenticatedRoute.tsx

```diff
@@ -0,0 +1,56 @@
+import { Route, redirect } from "@tanstack/react-router";
+import { Outlet } from "@tanstack/react-router";
+import axios, { AxiosError } from "axios";
+import { z } from "zod";
+import { NavBar } from "./components/navbar";
+import { rootRoute } from "./main";
+
+const userAuthStatusType = z.object({
+  message: z.string(),
+  redirect: z.string().optional(),
+  error: z.string().optional(),
+});
+
+export const authenticatedRoute = new Route({
```

Comment:
```suggestion
const authenticatedRoute = new Route({
```
Just to keep all the route imports alike, I suppose we could export authenticated route as a default export.